src/configure/services/source/sourceRepositoryService.ts

```diff
@@ -0,0 +1,126 @@
+                    remoteUrl: remoteUrl,
+                    branch: branch,
+                    commitId: commitId,
+                    localPath: repositoryPath
+                };
+            }
+            else if (GitHubProvider.isGitHubUrl(remoteUrl)) {
+                let repoId = GitHubProvider.getRepositoryIdFromUrl(remoteUrl);
+                return <GitRepositoryDetails>{
+                    sourceProvider: SourceProviderType.Github,
+                    repositoryId: repoId,
+                    repositoryName: repoId,
+                    remoteUrl: remoteUrl,
+                    branch: branch,
+                    commitId: commitId,
+                    localPath: repositoryPath
+                };
+            }
+            else {
+                throw new Error("Could not identify repository details. Ensure your git repo is managed with Azure Repos or Github");
+            }
+        }
+        else {
+            throw new Error("Remote repository is not configured. Manage your git repository with Azure Repos or Github");
+        }
+    }
+
+    /**
+     *
+     * @param pipelineYamlPath : local path of yaml pipeline in the extension
+     * @param context: inputs required to be filled in the yaml pipelines
+     * @returns: thenable object which resolves once all files are added to the repository
+     */
+    public addYmlFileToRepo(ymlFilePath: string, repoPath: string, context: any): Q.Promise<string> {
+        let deferred: Q.Deferred<string> = Q.defer();
+        fs.readFile(ymlFilePath, { encoding: "utf8" }, async (error, data) => {
+            if (error) {
+                throw new Error(error.message);
+            }
+            else {
+                let fileContent = Mustache.render(data, context);
+                let ymlFileUri = vscode.Uri.file(repoPath + "/azure-pipelines.yml");
+                fs.writeFileSync(ymlFileUri.fsPath, fileContent);
+                await vscode.workspace.saveAll(true);
+                deferred.resolve(ymlFileUri.fsPath);
+            }
+        });
+
+        return deferred.promise;
+    }
+
+    /**
+     * commits yaml pipeline file into the local repo and pushes the commit to remote branch.
+     * @param pipelineYamlPath : local path of yaml pipeline in the repository
+     * @returns: thenable object which resolves once commit is pushed to remote branch, and failure message if unsuccessful
+     */
+    public async commitAndPushPipelineFile(pipelineYamlPath: string): Promise<{ commitId: string, branch: string }> {
+        // TODO
+        // unstages the changes already staged
+        // stages the file at the mentioned path
+        // commit the staged file
+        // stage the previously staged changes
+
+
+        await this.gitReference.add(pipelineYamlPath);
+        let commit = await this.gitReference.commit("Add yml file to workspace", pipelineYamlPath);
```

Comment:
Should we show warning if there are more file that are staged

Reply:
Only changes to the files mentioned in the function are committed. Hence, we need not show warning as we do not commit other changes